src/SpinWaveTheory/SpinWaveTheory.jl

```diff
@@ -227,18 +225,18 @@ function swt_data_dipole(sys::System{0}, obs)
     # Precompute transformed exchange matrices and store in sys.interactions_union.
     for ints in sys.interactions_union
         for c in eachindex(ints.pair)
-            (; isculled, bond, scalar, bilin, biquad, general) = ints.pair[c]
-            i, j = bond.i, bond.j
+            (; bond, scalar, bilin, biquad, general) = ints.pair[c]
+            (; i, j) = bond
 
             if !iszero(bilin)  # Leave zero if already zero
                 J = Mat3(bilin*I)
-                bilin = S * (Rs[i]' * J * Rs[j]) 
+                bilin = sqrt(S[i]*S[j]) * Rs[i]' * J * Rs[j]
```

Comment:
I think it is a good idea to remove any `S`s in the definition of `billin` to support the varying-S spin wave theory.

Reply:
Also, removing `S`s from the definition would help my implementation of the non-perturbative calculation. It's imperative to maintain the vertex function's order in terms of `S`. Currently, I have to divide everything by `S`, which can be somewhat counterintuitive.

Reply:
Yes I agree.